pilot/pkg/model/config.go

```diff
@@ -331,6 +336,46 @@ func mostSpecificHostWildcardMatch[V any](needle string, wildcard map[host.Name]
 	return matchHost, matchValue, found
 }
 
+// OldestMatchingHost returns the oldest matching host for a given needle (whether specific or wildcarded)
+func OldestMatchingHost[V CreationTimestamper](needle host.Name, specific map[host.Name]V,
```

Comment:
Why can't we just take Config map[host.Name]Config as argument? Why does it need to be templatized with specific interface (CreationTimestamper)?

Reply:
I wanted to keep it generic like MostSpecificHostMatch. To keep it generic, you need an interface type constraint so I created CreationTimestamper

Reply:
The existing MostSpecificHostMatch is used for `set` of host names also. That is the reason we have made it generic. But in this specific case there may not be any such need - we always operate on Config here. NBD just pointing out

Reply:
Fair, I had to add the GetCreation Timestamp func to begin with. What I think I'll do is remove the generic and just make the args take the interface instead of a type param so that pointers are stored in the maps instead of copying large config objects around

Reply:
I was worried about large config specs being copied in the slice, but then I realized that the actual spec is `any` so no worries there. Changed it